### PR TITLE
Integration test for Model

### PR DESCRIPTION
**Pull request by mwielgus**
It is not a full E2E test (like in `integration/*`) but it checks whether the data is correctly passed from `Source` through `Cache` to `Model` and if `Model` returns the expected metrics properly. It doesn't bring up any instance on Kuberentes.

More tests will be added in follow-up PRs.

### CODE REVIEW

File: integration/model_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	loadAverageMilli = 300
+)
+
+func (t testSource) buildPods(start time.Time) []source_api.Pod {
+
+	timeElapsed := start.Sub(t.createTimestamp)
+
+	result := []source_api.Pod{}
+	for i := 0; i < podCount; i++ {
+
```
Comment (piosz) [inline integration/model_test.go]: nit:removing this empty line

File: integration/model_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	createTimestamp time.Time
+}
+
+const (
+	podCount         = 10
+	testNamespace    = "testnamespace"
+	loadAverageMilli = 300
+)
+
+func (t testSource) buildPods(start time.Time) []source_api.Pod {
```
Comment (piosz) [inline integration/model_test.go]: Does it make sense to use fuzzer here and then overwrite some fields required for the test?

File: integration/model_test.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		PodName:       "pod-0",
+		MetricRequest: model_api.MetricRequest{
+			Start:      start,
+			End:        time.Time{},
+			MetricName: "cpu-usage",
+		},
+	})
+	assert.NoError(err)
+	//TODO: Expect more than 1 metric once #551 is fixed
+	assert.True(len(metrics) >= 1)
```
Comment (piosz) [inline integration/model_test.go]: I believe there is a better function to test this condition. Same below.

File: integration/model_test.go (hunk 2 above)
Comment (mwielgus) [inline integration/model_test.go]: Fuzzed structs are already used in unit tests to check corner cases. Here I would like to check if nothing is lost/dropped on the way if the values are correct. The values set below are (mostly) needed and the rest is irrelevant from the point of view of this test.

File: integration/model_test.go (hunk 1 above)
Comment (mwielgus) [inline integration/model_test.go]: Done

File: integration/model_test.go (hunk 3 above)
Comment (mwielgus) [inline integration/model_test.go]: Not a big difference but OK. Changed to `NotEmpty` and `InEpsilon`.

**Comment by piosz** [comment]
LGTM feel free to merge on green

**Comment by mwielgus** [comment]
All green, merging.

**Comment by vishh** [comment]
LGTM
